test/unit/mocklib.py

```diff
@@ -250,7 +277,27 @@ def init_mocks():
     patchings remains active for the rest of the Python environment
     lifetime, the patchers' `stop` method is never called!
     """
-    patch('ncs.application.Application', new=mock.Mock).start()
-    patch('ncs.dp.Action.action', side_effect=lambda fn: fn).start()
-    patch('ncs.dp.Action.__init__', return_value=None).start()
+    sys.modules['ncs'] = Mock(application=Mock(Application=Mock),
+                              dp=Mock(Action=MockAction))
+    sys.modules['_ncs'] = Mock(LIB_VSN=0x07060000)
+    sys.modules['_ncs.dp'] = Mock(Action=Mock)
+    sys.modules['_ncs.maapi'] = Mock()
+    actions = ["setup-xmnr", "delete-state", "disable-state", "enable-state", "list-states",
```

Comment:
not sure if applicable, but might be worth grepping list of actions from `drned-xmnr.yang` to reduce this mock code maintenance for future updates?

Reply:
May be worth it; but "grepping" a YANG module is inherently unreliable, using pyang would be better.  But that means one more dependency - any other options?

Reply:
I went with the "grep" variant, in the end; pyang followed by xslt is possible, but it takes noticeable time - this should not be the case for unit tests.